servatrice/src/serversocketinterface.cpp

```diff
@@ -974,6 +975,48 @@ Response::ResponseCode ServerSocketInterface::cmdBanFromServer(const Command_Ban
     return Response::RespOk;
 }
 
+Response::ResponseCode ServerSocketInterface::cmdForgotPassword(const Command_ForgotPassword &cmd)
+{
+
+	bool forgotPasswordEnabled = settingsCache->value("users/enableForgotPassword", true).toBool();
+	if (!forgotPasswordEnabled)
+		return Response::RespInternalError;
+
+	if (tooManyForgotPasswordAttempts(this->getAddress()))
+		return Response::RespInternalError;
+
+	bool requireForgotPasswordEmail = settingsCache->value("users/forgotpasswordemailrequired", true).toBool();;
+	bool requireForgotPasswordClientID = settingsCache->value("users/forgotpasswordclientidrequired", true).toBool();
+
+	QString userName = QString::fromStdString(cmd.user_name());
+	QString password = QString::fromStdString(cmd.password());
+	QString userEmail = QString::fromStdString(cmd.email());
+	QString clientID = QString::fromStdString(cmd.clientid());
+
+	//if requesting user is a trusted user, change the password for the requested user
```

Comment:
I don't think "forgot password" should be what admin clients use to reset user passwords. They should use the "change password" function instead. An admin using this should just trigger the normal password reset process